pkg/controller/controller_broker.go

```diff
@@ -86,6 +86,7 @@ const (
 	errorProvisionCallFailedReason             string = "ProvisionCallFailed"
 	errorErrorCallingProvisionReason           string = "ErrorCallingProvision"
 	errorDeprovisionCalledReason               string = "DeprovisionCallFailed"
```

Comment:
I know it's out of scope of your PR, but shouldn't this field be called "errorDeprovisionCallFailedReason"?
"deprovision called" sounds like a success, not error :)
I noticed this because you have replaced it with the other one for when deprovision is blocked.

Reply:
@nilebox yeah, the name is janky.  I would like to go over all of these with a fine-toothed comb and will hit when we do that.